### ISSUE TITLE
clang: error: no such file or directory: 'src/scuff/taylor_duffy.cpp'

### ISSUE THREAD

**Issue by ubuntuslave** [DavidPowell/OpenModes#1]
Installing from source with

```
$ python3 setup.py install
```

I get the following error related to `scuff/taylor_duffy.cpp`:

```
compile options: '-I/usr/local/lib/python3.5/site-packages/numpy/core/include -I/usr/local/include -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/include/python3.5m -c'
clang++: src/scuff/TaylorDuffy.cc
src/scuff/TaylorDuffy.cc:585:7: warning: unused variable 'MaxMonomial' [-Wunused-variable]
  int MaxMonomial = TDW->MaxMonomial[WhichP];
      ^
1 warning generated.
clang++: src/scuff/taylor_duffy.cpp
clang: error: no such file or directory: 'src/scuff/taylor_duffy.cpp'
clang: error: no input files
clang: error: no such file or directory: 'src/scuff/taylor_duffy.cpp'
clang: error: no input files
error: Command "clang++ -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers -I/usr/local/lib/python3.5/site-packages/numpy/core/include -I/usr/local/include -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/include/python3.5m -c src/scuff/taylor_duffy.cpp -o build/temp.macosx-10.11-x86_64-3.5/src/scuff/taylor_duffy.o" failed with exit status 1
```

**Comment by DavidPowell** [comment]
Thanks for the report. Are you using the latest master version, or a tagged release?

**Comment by ubuntuslave** [comment]
I'm on the **master** branch. I just cloned it today from the repo.

**Comment by DavidPowell** [comment]
Should be fixed in master now.

**Comment by ubuntuslave** [comment]
That problem got fixed now.

However, I'm not sure if this is a separate issue related to linking against the `gomp` library, which I don't seem to have on OS X.

I'm getting the following error:

```
clang++ -bundle -undefined dynamic_lookup -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -arch x86_64 build/temp.macosx-10.11-x86_64-3.5/src/scuff/pcubature.o build/temp.macosx-10.11-x86_64-3.5/src/scuff/taylor_duffy.o build/temp.macosx-10.11-x86_64-3.5/src/scuff/TaylorDuffy.o -L/usr/local/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/sqlite/lib -lgomp -o build/lib.macosx-10.11-x86_64-3.5/openmodes/taylor_duffy.cpython-35m-darwin.so
ld: library not found for -lgomp
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ld: library not found for -lgomp
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by DavidPowell** [comment]
The library gomp is related to OpenMP support, which enables the critical fortran sections of code to run in parallel.

Unfortunately the compilation and linking options for OpenMP depend heavily on your compilers (both C and fortran) and platform. So far I have catered for gcc and gfortran or intel C and fortran compilers. Other compilers could be added if I know the correct options.

Googling reveals that recent versions of clang do include OpenMP support, but I'm not sure if this has been included into XCode yet.

If you just want to get it working, in setup.py line 59 remove the option "-fopenmp" and on line 60 remove the option "gomp" (but keep the empty square brackets).

Hopefully this will enable you to compile and run, at the expense of reduced performance.

If OpenMP is included in XCode (or when it does become available) I would be happy to include the appropriate options if a Mac user can help me figure them out.

**Comment by ubuntuslave** [comment]
Yes, that got it finally installed. 

Thanks a lot for the great help!. I'll keep my ears open the OpenMP addition to Xcode's clang.
